src/aggregator/aggregator/leader_flush_mgr.go

```diff
@@ -181,20 +190,69 @@ func (mgr *leaderFlushManager) Prepare(buckets []*flushBucket) (flushTask, time.
 		}
 	}
 
-	durationSinceLastPersist := time.Duration(nowNanos - mgr.lastPersistAtNanos)
-	if mgr.flushedSincePersist && durationSinceLastPersist >= mgr.flushTimesPersistEvery {
-		mgr.lastPersistAtNanos = nowNanos
-		mgr.flushedSincePersist = false
-		flushTimes := mgr.prepareFlushTimesWithLock(buckets, allShards)
-		mgr.flushTimesManager.StoreAsync(flushTimes)
-	}
-
 	if !shouldFlush {
+		// NB: if enough time has passed since last flush time persist, run the
+		// persist by itself.
+		if shouldUpdateFlushTimes {
+			mgr.flushTask.runNextOnCompleteTask = false
+			return mgr.flushTask.onCompleteTask, waitFor
+		}
+
 		return nil, waitFor
 	}
+
 	return mgr.flushTask, waitFor
 }
 
+type runFunctionTask struct {
```

Comment:
This wrapper probably unnecessary? And just need a `func()` and run if non-nil?

Reply:
Yeah, this was so we could return the shard flush times update from Prepare and allow the flush manager to run it, but with the upcoming refactor this won't be necessary